File: python3/Q09_palindrome_number.py
#
# Difficulty Level: EASY
#
# Given an integer x, return true if x is a palindrome, and false otherwise.
#

class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        
        number = x
        reverse = 0
        
        while number:
            reverse = reverse * 10 + number % 10
            number //= 10
        return x == reverse
    
    def isPalindrome2(self, x: int) -> bool:
        return False if x < 0 else str(x)[::-1] == str(x)

File: java/Q09_palindrome_number.java
/*
 * Difficulty Level: EASY
 * 
 * Given an integer x, return true if x is a palindrome, and false otherwise.
 * 
 */

public class Q09_palindrome_number {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        int number = x;
        int reverse = 0;

        while (x > 0) {
            reverse = (reverse * 10) + (x % 10);
            x /= 10;
        }

        return number == reverse;
    }
}
